src/schoology-extractor/tests/test_lib/test_request_client.py

```diff
@@ -0,0 +1,231 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pytest
+
+from data_extractor.lib.request_client import RequestClient
+
+pytest.fake_key = 'TEST_KEY'
+pytest.fake_secret = 'TEST_SECRET'
+pytest.fake_endpoint_url = 'FAKE_URL'
+pytest.default_url = 'https://api.schoology.com/v1/'
+
+
+@pytest.fixture
+def default_request_client():
+    return RequestClient(pytest.fake_key, pytest.fake_secret)
+
+
+class TestRequestClient:
+
+
+    class Test_when_constructing:
+        def test_given_not_passing_in_a_url_then_use_the_default_url(self):
+            request_client = RequestClient(pytest.fake_key, pytest.fake_secret)
+            assert request_client.base_url == pytest.default_url
+
+        def test_given_custom_url_then_use_it_for_the_base_url(self):
+
+            # Arrange
+            custom_url = "a_custom_url"
+
+            # Act
+            request_client = RequestClient(
+                pytest.fake_key, pytest.fake_secret, custom_url)
+
+            # Assert
+            assert request_client.base_url == custom_url
+
+
+    class Test_when_building_request_header:
+        def test_then_consumer_key_is_present(self, default_request_client):
+            assert self._present_in_header(
+                default_request_client._request_header,
```

Comment:
You have default_request_client here hiding the function of the same name in outer scope.  Other places too.

Reply:
Actually, it is a test fixture, by having the same name, I am kind of "injecting" the result of the default_request_client() function.

More information [here](https://docs.pytest.org/en/stable/fixture.html#fixtures-as-function-arguments)